cores/arduino/gd32/pinmap.c

```diff
@@ -62,11 +62,11 @@ void pin_function(PinName pin, int function)
     }
 
     uint32_t mode   = GD_PIN_MODE_GET(function);
-    uint32_t af     = GD_PIN_AF_GET(function);
+    // uint32_t af  =   GD_PIN_AF_GET(function);
```

Comment:
This can't be correct, the variable is used in the codepath below (`if defined(GD32F3x0) || defined(GD32F1x0) || defined(GD32F4xx)`), compilation for fail for all these variants. Maybe we shoud `(void) unusedVar` or only declare them if in the `#if` block they're used -- but this breaks the nice "get all sub-info from function" block.

Reply:
and this is why I request reviews. *facepalm* Thank you.

Reply:
This looks correct now.